src/main/java/com/squareup/catalog/demo/example/ListDiscountsExample.java

```diff
@@ -15,72 +15,83 @@
 import com.squareup.catalog.demo.util.Moneys;
-import com.squareup.connect.ApiException;
-import com.squareup.connect.api.CatalogApi;
-import com.squareup.connect.api.LocationsApi;
-import com.squareup.connect.models.CatalogDiscount;
-import com.squareup.connect.models.CatalogObject;
-import com.squareup.connect.models.ListCatalogResponse;
-
-import static com.squareup.connect.models.CatalogObjectType.DISCOUNT;
+import com.squareup.square.api.CatalogApi;
+import com.squareup.square.api.LocationsApi;
+import com.squareup.square.models.CatalogDiscount;
+import com.squareup.square.models.CatalogObject;
 
 /**
  * This example lists all discounts in the catalog.
  */
 public class ListDiscountsExample extends Example {
 
+  private String cursor = null;
+
   public ListDiscountsExample(Logger logger) {
     super("list_discounts", "List all discounts.", logger);
   }
 
   @Override
-  public void execute(CatalogApi catalogApi, LocationsApi locationsApi) throws ApiException {
-    String cursor = null;
+  public void execute(CatalogApi catalogApi, LocationsApi locationsApi) {
+
+    // Optional parameters can be set to null.
+    Long catalogVersion = null;
+
     do {
       // Retrieve a page of discounts.
-      ListCatalogResponse listResponse = catalogApi.listCatalog(cursor, DISCOUNT.toString());
-      if (checkAndLogErrors(listResponse.getErrors())) {
-        return;
-      }
-
-      if (listResponse.getObjects().isEmpty() && cursor == null) {
-        logger.info("No discounts found.");
-        return;
-      }
+      catalogApi.listCatalogAsync(cursor, CatalogObjectTypes.DISCOUNT.toString(), catalogVersion).thenAccept(result -> {
+        if (checkAndLogErrors(result.getErrors())) {
+          return;
+        }
 
-      for (CatalogObject discountObject : listResponse.getObjects()) {
-        CatalogDiscount discount = discountObject.getDiscountData();
-        String amount = null;
+        List<CatalogObject> discounts = result.getObjects();
+        if (discounts == null || discounts.size() == 0) {
+          if (cursor == null) {
+            logger.info("No discounts found.");
+            return;
+          }
+        } else {
+          for (CatalogObject discountObject : discounts) {
+            CatalogDiscount discount = discountObject.getDiscountData();
+            String amount = null;
 
-        // Determine which type of discount this is.
-        switch (discount.getDiscountType()) {
-          case FIXED_AMOUNT:
-            amount = Moneys.format(discount.getAmountMoney());
-            break;
-          case FIXED_PERCENTAGE:
-            amount = discount.getPercentage() + "%";
-            break;
-          case VARIABLE_AMOUNT:
-            amount = "variable $";
-            break;
-          case VARIABLE_PERCENTAGE:
-            amount = "variable %";
-            break;
-        }
+            // Determine which type of discount this is.
+            switch (DiscountTypes.valueOf(discount.getDiscountType())) {
```

Comment:
I used `valueOf()` because `discount.getDiscountType()` returns a String and I wanted to convert it to the DiscountType enum type. I thought that using enum types as part of the switch-case statement would be a better design - feels a bit neater than hardcoding 4 strings. If you don't like it we can change it.